src/go/rpk/pkg/cli/cmd/cluster.go

```diff
@@ -54,5 +55,7 @@ func NewClusterCommand(fs afero.Fs) *cobra.Command {
 	offsets.Use = "offsets"
 	command.AddCommand(offsets)
 
+	command.AddCommand(config.NewConfigCommand(fs))
```

Comment:
I was hoping to deprecate the `cluster` subcommand, having everything split into either `debug` or `redpanda` directly (and to hide the start/stop/etc. within `redpanda)`. cc @vsaraswat here. It's not clear today why I'd want `cluster` vs. `redpanda`, and I'd like to iron out that distinction here before we introduce more into `cluster` if we are planning to do something else shortly.

Also, I'd like to figure out how this relates / fits with broker configuration modification through the `IncrementalAlterConfigs` request. I don't think we currently support this today, but I think we should: it's the standard within the ecosystem.

Reply:
@twmb what would you recommend for as command for config if we deprecate cluster? Would it be `redpanda config` or just top level `config`?

Reply:
Today we have the following under `rpk redpanda`: `[admin check config mode start stop tune]`

- admin is for talking to the admin port, and likely could be better arranged (`admin config` for configuration modification, `admin brokers` for broker modification).
- check, which does a whole lot of local benchmarking without warning. This should probably live under `debug`.
- config to modify the broker configuration, what this PR hopefully sunsets
- mode which modifies the broker configuration, which this PR hopefully sunsets / can be added on to this PR in a way, if we want to keep "mode" switches
- start/stop, for literally starting and stopping redpanda. These potentially could be hidden commands, since this is what systemd itself invokes per our systemd file
- tune, which runs tuners which modify your system _for_ redpanda, and aren't related to actually running redpanda. This potentially should live under `debug`.

So, moving start/stop/mode/tune/check, and hopefully having config replaced with this, all that remains is `admin`. I know when I first used rpk, I was confused how / why cluster was different from "redpanda". Since this is modifying the configuration of multiple redpanda servers, should this live within the redpanda config?

Basically I think we have too many ways to modify multiple things, and while we may think that's really useful at times for some niche reasons, my suspicion is it's not all too clear for end users.

Reply:
I think we discussed this on Slack a while back, but let me lay out my case for this naming for the record here:

My take on naming for commands that mutate redpanda in non-kafka ways is that the important distinction is between the node and cluster.  Node commands are things like start/stop a service, write a redpanda.yml config.  Cluster commands operate on the logical redpanda cluster, i.e. operations that result in changes in the redpanda controller topic, and don't change the local state of any nodes.  We have relatively few `cluster` commands but that's in large part because the admin API functionality is quite new and just hasn't bubbled up into the CLI yet.

In this particular case, the cluster/node distinction is particularly salient because nodes have local config which is disjoint from the cluster config.

### Why not `admin` as a prefix?

Editing a config option isn't intrinsically more _administrative_ than editing an ACL or creating a topic.  What we really mean by admin is that the command flows through a REST API rather than Kafka -- that's an implementation detail that doesn't help make the command name clearer to the user.

### Why not `redpanda` as a prefix?

Everything rpk does is about redpanda -- to understand why some commands are redpanda and some aren't, one has to understand the underlying distinction between kafka-general stuff and redpanda-specific stuff.  What we really mean by `redpanda` in the existing commands is that we're modifying our daemons, rather than modifying logical kafka entities.  It's all redpanda.